server/BudgetTracker.Business/Budgeting/DerivedBudgetAttributes.cs

```diff
@@ -0,0 +1,26 @@
+using BudgetTracker.Common.Models;
+
+namespace BudgetTracker.Business.Budgeting
+{
+    public class DerivedBudgetAttributes
+    {
+        public static bool IsBudgetPercentBasedBudget(Budget budget)
```

Comment:
I think this is a big overdoing it, IMO we can just make the comparison normally.

Reply:
This makes it explicit. Sometimes, it can get confusing if you see the computation... it might take a minute to realize what `if (budget.ParentBudgetId == null)` actually means. That can easily be mistaken to be thought of as a simple null check to catch errors and not an actual business rule. This way, we essentially put a name to the operation so the reader knows immediately what's going on. It's self documenting code.

Also, in the future, we may want to add more logic to see if a budget is a root budget (not likely, but this paves the way to make that process easy.

Reply:
Sounds good not against it just not how I would have done it, fine with going forward with this.